Apple/NetworkExtension/PacketTunnelProvider.swift

```diff
@@ -1,7 +1,22 @@
+import libburrow
 import NetworkExtension
+import OSLog
 
 class PacketTunnelProvider: NEPacketTunnelProvider {
+    let logger = Logger(subsystem: "com.hackclub.burrow", category: "General")
     override func startTunnel(options: [String: NSObject]?, completionHandler: @escaping (Error?) -> Void) {
+        let fild = libburrow.retrieve()
+        if fild == -1 {
+            // Not sure if this is the right way to return an error
```

Comment:
It is okay for now. Error reporting should be on the Rust side of things, and but Swift needs an error to return to the completion handler.

Reply:
The job of this code should be to pass an error to the system if anything in the tunnel setup process fails. Swift cannot really do anything with the file descriptor, so we should not write code to retrieve it into Swift. What this should look like long term is likely something like this:

```
Apple Extension:
1. Swift – PacketTunnelProvider.startTunnel
2. Swift – func start() throws
2.5 – any FFI layers
3. Rust – start() -> Result<Void, Error>
    - It calls a function to find the interface
    - It calls configure() to configure the interface
    - It spawns a tokio runtime
    - It spawns a task on that tokio runtime to run burrow's core loop
    - It returns a handle to said task

Linux Daemon:
1. tokio::main function
1. clap to parse arguments
2. create new TunInterface
3. call configure() to configure the interface
4. it runs burrow's core loop

Linux CLI:
1. tokio::main function
2. clap to parse arguments
3. Looks for unix socket for Linux daemon
4. Sends commands, daemon establishes tunnel
5. Exits
```

Reply:
got it, that makes sense! 
For the Apple Extension, I suppose we would have to either make the main burrow loop outrun the execution of the start function, or use something like a Swift [task](https://docs.swift.org/swift-book/documentation/the-swift-programming-language/concurrency/#Tasks-and-Task-Groups) to call a blocking Rust function? For the first case, I'm not quite sure how to achieve that in Rust, as the Tokio runtime would have to be dropped when the function returns. Perhaps calling mem::forget on the tokio runtime would work?